### PR TITLE
fix: Do not log successful health and alive requests via exludes.

### PR DESCRIPTION
**Pull request by merlin-northern**
Changelog: Title
Ticket: None
Signed-off-by: Peter Grzybowski <peter@northern.tech>

### CODE REVIEW

**Comment by merlin-northern** [comment]
> I think is a good idea.. But donno about completely excluding that, maybe for later debug purposes we can need to check that? Why not logging them in a DEBUG/TRACE level so to enable if needed?

logging via debug commit: https://github.com/mendersoftware/go-lib-micro/pull/180/commits/dea3cbf5313d9c3adc3f72e909e849c35fbfba3c
now we can choose which one we want.

**Review by oldgiova — APPROVED** [review]

**Comment by merlin-northern** [comment]
@MaciejTe @tranchitella what do you think, which way shall we go (Alex already expressed his opinion on slack, that we should log to debug (second commit))?

File: accesslog/middleware.go [hunk 1]
```diff
@@ -57,18 +57,37 @@ func (mw *AccessLogMiddleware) MiddlewareFunc(h rest.HandlerFunc) rest.HandlerFu
...
+				"type":         TypeHTTP,
+				"ts":           util.StartTime().Round(0),
+				"status":       util.StatusCode(),
+				"responsetime": util.ResponseTime().Seconds(),
+				"byteswritten": util.BytesWritten(),
+				"method":       r.Method,
+				"path":         r.URL.Path,
+				"qs":           r.URL.RawQuery,
+			}).Print(mw.executeTextTemplate(util))
+		}
```
Comment (tranchitella) [inline accesslog/middleware.go:90]:
```suggestion
		logger := requestlog.GetRequestLogger(r)
		logged := false
		log := logger.WithFields(logrus.Fields{
			"type":         TypeHTTP,
			"ts":           util.StartTime().Round(0),
			"status":       util.StatusCode(),
			"responsetime": util.ResponseTime().Seconds(),
			"byteswritten": util.BytesWritten(),
			"method":       r.Method,
			"path":         r.URL.Path,
			"qs":           r.URL.RawQuery,
		})
		for pathSuffix, status := range DebugLogsByPathSuffix {
			if util.StatusCode() == status && strings.HasSuffix(r.URL.Path, pathSuffix) {
				log.Debug(mw.executeTextTemplate(util))
				logged = true
			}
		}

		if !logged {
			log.Print(mw.executeTextTemplate(util))
		}
```

**Review by tranchitella — COMMENTED** [review]
Do we need to repeat the `WithFields` call?

**Comment by merlin-northern** [comment]
> Do we need to repeat the `WithFields` call?

fixed.

File: accesslog/middleware.go [hunk 2]
```diff
@@ -68,7 +69,17 @@ func (mw *AccessLogMiddleware) MiddlewareFunc(h rest.HandlerFunc) rest.HandlerFu
 			"method":       r.Method,
 			"path":         r.URL.Path,
 			"qs":           r.URL.RawQuery,
-		}).Print(mw.executeTextTemplate(util))
+		})
+		for pathSuffix, status := range DebugLogsByPathSuffix {
+			if util.StatusCode() == status && strings.HasSuffix(r.URL.Path, pathSuffix) {
+				log.Debug(mw.executeTextTemplate(util))
+				logged = true
```
Comment (tranchitella) [inline accesslog/middleware.go:76]: minor optimization, you can `break` here

File: accesslog/middleware_gin.go [hunk 3]
```diff
@@ -128,7 +128,17 @@ func Middleware() gin.HandlerFunc {
 		l = l.F(logCtx)
 
 		if code < 400 {
-			l.Info()
+			logged := false
+			for pathSuffix, status := range DebugLogsByPathSuffix {
+				if code == status && strings.HasSuffix(c.Request.URL.Path, pathSuffix) {
+					l.Debug()
+					logged = true
```
Comment (tranchitella) [inline accesslog/middleware_gin.go:135]: minor optimization, you can `break` here

**Review by tranchitella — APPROVED** [review]
Pre-approving.

**Pull request merged**